src/pages/HomePage/HomePage.tsx

```diff
@@ -1,18 +1,66 @@
 import React from 'react';
-// import { Button } from '../../components/Button/Button';
+import useUserInfo from '../../hooks/useUserInfo';
+import { Button } from '../../components/Button/Button';
 import { Header } from '../../components/Header/Header';
 import { Menu } from '../../components/Menu/Menu';
 import { ContainerMain } from '../../components/ContainerMain/ContainerMain';
 import { Submenu } from '../../components/Submenu/Submenu';
+import {
+  GoogleLogin,
+  GoogleLoginResponse,
+  GoogleLoginResponseOffline,
+  GoogleLogout,
+} from 'react-google-login';
+import { UserProfile } from '../../models/UserProfile';
+
+export interface LoginState {
+  user: UserProfile | null;
+}
+
+const initialState: LoginState = { user: null };
+
+export const CurrentUserContext = React.createContext<LoginState>(initialState);
 
 export const HomePage: React.FC = ({}) => {
+  const { userInfo, dispatchUserInfo } = useUserInfo(initialState);
+
+  const responseGoogle = (
```

Comment:
you forgot about error handling, try to open sign in google window and then close it without confirmation for example.

Reply:
I use ` onFailure={(response) => console.log(response)}` at string 46. I should put it inside responseGoogle function, rigth? @bondarchukse

Reply:
it's ok to set error message to the console for now, but needs to add TODO comment to handle this case in future